react-hot-button/my-app/src/App.tsx

```diff
@@ -0,0 +1,43 @@
+// import { useState } from 'react'
+import './ButtonContainer.css';
+import './App.css';
+import { ButtonContainer } from './ButtonContainer';
+import { ClickCount } from './ClickCount';
+import { useState } from 'react';
+
+const buttonText = 'Hot Button';
+let className = 'cold';
+function App() {
+  const [count, setCount] = useState(0);
+
+  function handleClick() {
+    setCount(count + 1);
+    if (count === 3) {
+      className = 'cool';
```

Comment:
You are setting className here in the handleClick function, so its value gets lost when handleClick returns. It was clever of you to solve this by making `className` a global variable (by putting it _outside_ of App). However, that's considered bad practice because it can introduce bugs.

For example, suppose you had 2 Hot Buttons on the same page. They would both share the _same_ value of `className`, so they would be displaying the incorrect color.

There are two ways to solve this. One, you could make `className` be state. That way it retains its value across renders. The other is to do this computation outside of handleClick, just right inside the App function.

If you had done layout followed by state followed by events, you would have found this problem.